Sources/SwiftBSON/BSONDocument.swift

```diff
@@ -517,3 +517,42 @@ extension BSONDocument: BSONValue {
 extension BSONDocument: CustomStringConvertible {
     public var description: String { self.toExtendedJSONString() }
 }
+
+extension BSONDocument {
+    /// Helper function to check for BSON document equality regardless of the order of key-value pairs
+    internal func equalsIgnoreOrderHelper(a: BSONDocument, b: BSONDocument) -> Bool {
+        let keysA = a.keys.sorted()
+        let keysB = b.keys.sorted()
+
+        guard keysA == keysB else {
```

Comment:
Rather than sorting and checking for equality (which is expensive), we can simply check to see if their lengths are the same. The next portion will ensure that the keys and values are all equal in one step.